### ISSUE TITLE
LDAP - Can't login after upgrade 10.0.8

### ISSUE THREAD

**Issue by Selven99** [glpi-project/glpi#15162]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.8

### Bug description

Hello,

After upgrade to 10.0.8, we cannot login with LDAP directory.

Configuration is :
![image](https://github.com/glpi-project/glpi/assets/11805144/34ab51ea-1ea4-40d0-a6cb-20747179f4cd)

Without bind user/password, there is an exception here:
```
object(TypeError)#62 (7) { ["message":protected]=> string(73) "ldap_search(): Argument #1 ($ldap) must be of type LDAP|array, bool given" ["string":"Error":private]=> string(0) "" ["code":protected]=> int(0) ["file":protected]=> string(30) "/var/www/html/src/AuthLDAP.php" ["line":protected]=> int(3537) ["trace":"Error":private]=> array(7) { [0]=> array(4) { ["file"]=> string(30) "/var/www/html/src/AuthLDAP.php" ["line"]=> int(3537) ["function"]=> string(11) "ldap_search" ["args"]=> array(4) { [0]=> bool(true) [1]=> string(20) "dc=my-domain,dc=tld" [2]=> string(12) "(uid=myuser)" [3]=> array(2) { [0]=> string(2) "dn" [1]=> string(3) "uid" } } } [1]=> array(6) { ["file"]=> string(26) "/var/www/html/src/Auth.php" ["line"]=> int(261) ["function"]=> string(12) "searchUserDn" ["class"]=> string(8) "AuthLDAP" 
```

`src/Auth.php` : 
```php
            try {
                $infos = AuthLDAP::searchUserDn($this->ldap_connection, [
                    'basedn'            => $ldap_method['basedn'],
                    'login_field'       => $ldap_method['login_field'],
                    'search_parameters' => $params,
                    'user_params'       => [
                        'method' => AuthLDAP::IDENTIFIER_LOGIN,
                        'value'  => $login
                    ],
                    'condition'         => Sanitizer::unsanitize($ldap_method['condition']),
                    'user_dn'           => $this->user_dn
                ]);
            } catch (\Throwable $e) {
                ErrorHandler::getInstance()->handleException($e, true);
                $this->addToError(__('Unable to connect to the LDAP directory'));
                return false;
            }
```

`$this->ldap_connection` is a boolean, not an LDAP array.

In my case, we are not using bind credentials because we use SSL certificates for authentication. But after enabling login with bind user and password, everything works perfectly!

Note : login with GLPI database works fine. Only LDAP without bind seems to be affected.

Thank you!

### Steps To reproduce

1. Create a LDAP Auth directory
2. Choose "No" for "Use Bind"
3. Validate
4. Try to login

--> You will trigger the catch and error message: "Unable to connect to the LDAP directory"

### Your GLPI setup information

Operating system: Linux 6366dd0c922a 5.15.49-linuxkit #1 SMP PREEMPT Tue Sep 13 07:51:32 UTC 2022 aarch64
PHP 8.2.7 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apcu, ctype, curl, date,
	dom, exif, fileinfo, filter, ftp, gd, hash, iconv, imap, intl, json, ldap, libxml, mbstring, mysqli, mysqlnd, openssl, pcre,
	pdo_sqlite, posix, random, session, sodium, sqlite3, standard, tokenizer, xml, xmlreader, xmlwriter, zlib)
Setup: max_execution_time="30" memory_limit="128M" post_max_size="25M" safe_mode="" session.save_handler="files"
	upload_max_filesize="25M" 
Software: Apache/2.4.56 (Debian) (Apache/2.4.56 (Debian) Server at localhost Port 80
)
	Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
Server Software: mariadb.org binary distribution
	Server Version: 10.5.21-MariaDB-1:10.5.21+maria~ubu2004
	Server SQL Mode: STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
	Parameters: glpi@db/glpi
	Host info: db via TCP/IP
	
PHP version (8.2.7) is supported.PHP version (8.2.7) is supported.
Sessions configuration is OK.Sessions configuration is OK.
Allocated memory is sufficient.Allocated memory is sufficient.
mysqli extension is installed.mysqli extension is installed.
Following extensions are installed: dom, fileinfo, filter, libxml, json, simplexml, xmlreader, xmlwriter.Following extensions are installed: dom, fileinfo, filter, libxml, json, simplexml, xmlreader, xmlwriter.
curl extension is installed.curl extension is installed.
gd extension is installed.gd extension is installed.
intl extension is installed.intl extension is installed.
zlib extension is installed.zlib extension is installed.
The constant SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES is present.The constant SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES is present.
Database engine version (10.5.21) is supported.Database engine version (10.5.21) is supported.
No files from previous GLPI version detected.No files from previous GLPI version detected.
The log file has been created successfully.The log file has been created successfully.
Write access to /var/www/html/files/_cache has been validated. Write access to /var/www/html/config has been validated. Write access to /var/www/html/files/_cron has been validated. Write access to /var/www/html/files has been validated. Write access to /var/www/html/files/_dumps has been validated. Write access to /var/www/html/files/_graphs has been validated. Write access to /var/www/html/files/_lock has been validated. Write access to /var/www/html/files/_pictures has been validated. Write access to /var/www/html/files/_plugins has been validated. Write access to /var/www/html/files/_rss has been validated. Write access to /var/www/html/files/_sessions has been validated. Write access to /var/www/html/files/_tmp has been validated. Write access to /var/www/html/files/_uploads has been validated.Write access to /var/www/html/files/_cache has been validated.
Write access to /var/www/html/config has been validated.
Write access to /var/www/html/files/_cron has been validated.
Write access to /var/www/html/files has been validated.
Write access to /var/www/html/files/_dumps has been validated.
Write access to /var/www/html/files/_graphs has been validated.
Write access to /var/www/html/files/_lock has been validated.
Write access to /var/www/html/files/_pictures has been validated.
Write access to /var/www/html/files/_plugins has been validated.
Write access to /var/www/html/files/_rss has been validated.
Write access to /var/www/html/files/_sessions has been validated.
Write access to /var/www/html/files/_tmp has been validated.
Write access to /var/www/html/files/_uploads has been validated.

Web server root directory configuration seems safe.Web server root directory configuration seems safe.
Sessions configuration is secured.Sessions configuration is secured.
OS and PHP are relying on 64 bits integers.OS and PHP are relying on 64 bits integers.
exif extension is installed.exif extension is installed.
ldap extension is installed.ldap extension is installed.
openssl extension is installed.openssl extension is installed.
Following extensions are installed: Phar. Following extensions are not present: bz2, zip.Following extensions are installed: Phar.
Following extensions are not present: bz2, zip.
Zend OPcache extension is installed.Zend OPcache extension is installed.
Following extensions are installed: ctype, iconv, mbstring, sodium.Following extensions are installed: ctype, iconv, mbstring, sodium.

Warning:  mkdir(): Permission denied in /var/www/html/src/Toolbox.php on line 1208

The directory could not be created in /var/www/html/marketplace.The directory could not be created in /var/www/html/marketplace.
Access to timezone database (mysql) is not allowed.Access to timezone database (mysql) is not allowed.

**Comment by cedric-anne** [comment]
Hi,

Could you check #15104 ? I made some changes on this piece of code that can fix this issue.

**Comment by Selven99** [comment]
Hello,

Thank you!

I just tried your fix: 
* WIth bind credentials: works fine, as usual
* Without bind: blank page, weird! No error 🤔 

Thank you!

**Comment by Selven99** [comment]
Hi,

It seems fixed, thank you! 🎉 
Everything works fine in both cases, with and without bind ❤️

**Comment by Selven99** [comment]
Hi @cedric-anne,

It is fixed with #15104 and #15188. But #15104 seems not to be in 10.0.9 release?
I tested with 10.0.9 and problem still here, I can't login without bind 😬 

Thank you!

**Comment by cedric-anne** [comment]
> #15104 seems not to be in 10.0.9 release?

Indeed, it is not part of the 10.0.9 release as we thought that #15188 was the only regression in GLPI 10.0.8.

#15104 is a partial rewrite of LDAP auth and will be part of GLPI 10.0.10.
